### PR TITLE
`assert_iamframe_equals()` passes if all-nan-cols are present

### LINKED ISSUE #515: `assert_iamframe_equal()` fails on meta-columns with all-nan values
The testing function `assert_iamframe_equal()` fails when comparing two `IamDataFrame` instances where one side has a meta-column with all `nan` values and the other side doesn't have that columns. Arguably, all-nan should be treated as identical to "column doesn't exist".

```python
df.meta.dropna(axis="columns", how="all", inplace=True)
```

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- ~Name of contributors Added to AUTHORS.rst~
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

Currently, `assert_iamframe_equal()` fails if one of the IamDataFrames has a meta-column of all-nan's (and the column doesn't exist in the other IamDataFrame). Arguably, all-nan should be treated as identical to "column doesn't exist".

The PR also adds a page for the testing module to the docs.

Closes #515

### CODE REVIEW

File: pyam/testing.py [hunk 1]
```diff
@@ -2,10 +2,36 @@
...
-    pdt.assert_frame_equal(a.meta, b.meta, check_dtype=False, check_like=True)
+    pdt.assert_frame_equal(
+        _drop_nan_col(left.meta),
+        _drop_nan_col(right.meta),
+        check_dtype=False,
+        check_like=True,
+    )
+
+
+def _drop_nan_col(df):
```
Comment (phackstock) [inline pyam/testing.py:36]: Does that need to be its own function? I would say a single line would also nicely fit in `pdt.assert_frame_equal` above.

**Review by phackstock — APPROVED** [review]
Looks good to me.
Just one small comment and it seems there some conflict in the README that needs to be resolved.

File: pyam/testing.py (hunk 1 above)
Comment (danielhuppmann) [reply pyam/testing.py]: You're right, changed as you suggested...

**Comment by phackstock** [comment]
Perfect, good to merge from my side.

**Pull request merged**